src/Jh/DataImportMagento/Writer/ShipmentWriter.php

```diff
@@ -64,6 +70,17 @@ public function writeItem(array $item)
                 ->addObject($shipment->getOrder())
                 ->save();
 
+            if (isset($item['tracking_number'], $item['carrier'])) {
+                $tracking = clone $this->trackingModel;
+                $tracking
+                    ->setShipment($shipment)
+                    ->setData('title', $item['carrier'])
+                    ->setData('number', $item['tracking_number'])
+                    ->setData('carrier_code', 'custom')
+                    ->setData('order_id', $shipment->getData('order_id'))
+                    ->save();
+                \Mage::dispatchEvent('after_shipping_data_save', ['item' => $item]);
```

Comment:
This does not work for me at all, this is a hack to get what you need to work. We should not be triggering events here. You need a better way to handle this. Maybe you can accept a `callable` in the constructor like `callable $shipmentEmailProcessor` then you can just call that with the shipment. 

```php
$tracking
    ->setShipment($shipment)
    ...

if (options['send-emails']) {
    $emailProcessor = $this->emailProcessor;
    $emailProcessor($tracking);
}

```

Then you can just build your object like:
```
 return new ShipmentWriter(
            $orderModel,
            $transaction,
            $trackingModel,
            $logger,
            function (Track $track) {
                 ... do your email sending here
            }
        );
```
What do you think? (pseudo code obviously)